src/dbally/similarity/elastic_vector_search.py

```diff
@@ -0,0 +1,170 @@
+        - Creates a new trained model with the specified model ID.
+        - Polls the model definition status until the model is fully defined.
+        - Starts the deployment of the model with a specified number of allocations.
+        - Polls the deployment status until the model deployment is started.
+        - Sets up an ingest pipeline using the deployed model.
+
+        """
+        try:
+            await self.client.ml.delete_trained_model(model_id=".elser_model_2", force=True)
+            print("Model deleted successfully, We will proceed with creating one")
+        except exceptions.NotFoundError:
+            print("Model doesn't exist, but We will proceed with creating one")
+
+        await self.client.ml.put_trained_model(model_id=".elser_model_2", input={"field_names": ["text_field"]})
+
+        while True:
+            status = await self.client.ml.get_trained_models(model_id=".elser_model_2", include="definition_status")
+
+            if status["trained_model_configs"][0]["fully_defined"]:
+                print("ELSER Model is downloaded and ready to be deployed.")
+                break
+            print("ELSER Model is downloaded but not ready to be deployed.")
+            time.sleep(5)
+
+        await self.client.ml.start_trained_model_deployment(
+            model_id=".elser_model_2", number_of_allocations=1, wait_for="starting"
+        )
+
+        while True:
+            status = await self.client.ml.get_trained_models_stats(
+                model_id=".elser_model_2",
+            )
+            if status["trained_model_stats"][0]["deployment_stats"]["state"] == "started":
+                print("ELSER Model has been successfully deployed.")
+                break
+            print("ELSER Model is currently being deployed.")
+            time.sleep(5)
+
+        await self.client.ingest.put_pipeline(
+            id="elser-ingest-pipeline",
+            description="Ingest pipeline for ELSER",
+            processors=[
+                {
+                    "inference": {
+                        "model_id": ".elser_model_2",
+                        "input_output": [{"input_field": "column", "output_field": "column_embedding"}],
+                    }
+                }
+            ],
+        )
+
+    async def store(self, data: List[str]) -> None:
+        """
+        Stores the given data in an Elasticsearch store.
+
+        This function performs the following steps:
+
+        1. Deploys the ELSER model by calling `deploy_elser_mode`.
+        2. Defines the mappings for the Elasticsearch index.
+        3. Deletes the existing index if it exists.
+        4. Creates a new index with the specified mappings and settings.
+        5. Stores the provided data in the Elasticsearch index using the default ingest pipeline.
+
+        Args:
+            data: The data to store in the Elasticsearch index.
+        """
+        await self.deploy_elser_model()
```

Comment:
Why we are deploying elser model? I think it should be responsibility of the user - maybe even user already has model deployed, or has some other orchiestration to do this.